tests/func/test_add.py

```diff
@@ -57,6 +58,25 @@ def test_add(tmp_dir, dvc):
     }
 
 
+def test_add_executable(tmp_dir, dvc):
+    tmp_dir.gen("foo", "foo")
+    st = os.stat("foo")
+    os.chmod("foo", st.st_mode | stat.S_IEXEC)
```

Comment:
@efiop this test fails on Windows because it's not possible to set `S_IEXEC` bit as mentioned in `os.chmod` docs:

https://docs.python.org/3/library/os.html#os.chmod

> Note Although Windows supports chmod(), you can only set the file’s read-only flag with it (via the stat.S_IWRITE and stat.S_IREAD constants or a corresponding integer value). All other bits are ignored.

Should we just skip this test on Windows?

Reply:
@dudarev Sure, but I'm more worried about `dvc add` not preserving exec bit https://github.com/iterative/dvc/pull/5061#issuecomment-748525155 . I've pushed the fix that fixed the `add` and now I see that checkout test broke, it might've been working by accident :slightly_frowning_face:

Reply:
Thinking about the windows, tests shouldn't be skipped, but actually we should check that things are working as expected even for windows.

Reply:
In the checkout test I was testing with `os.access(self.EXECUTABLE, os.X_OK)`. Maybe on Windows `os.access` returns True in this case.

For testing scenarios on Windows. Since it's not possible to set `S_IEXEC`, the expected behavior is probably: if .dvc file has `isexec: True` in it, the file still remains not executable on Windows after checkout. Or rather it remains the same as without that `isexec` property.